### PR TITLE
refactor: Modify the cache with a `merge` method to ensure no important data gets overwritten

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

This pull request ensures that data got in, for example, the  `GUILD_CREATE` event does not get lost when getting the same guild via HTTP call

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [ ] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [x] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [x] Breaking change
  - [x] New feature/enhancement
  - [x] Bugfix

### CODE REVIEW

File: interactions/api/http/channel.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ async def get_channel_messages(
         if isinstance(request, list):
             for message in request:
                 if message.get("id"):
-                    self.cache[Message].add(Message(**message))
+                    self.cache[Message].merge(Message(**message))
```
Comment (Toricane) [inline interactions/api/http/channel.py]:
```suggestion
                    self.cache[Message].merge(Message(**message, _client=self))
```

File: interactions/api/http/guild.py [hunk 2]
```diff
@@ -588,7 +585,8 @@ async def add_guild_member(
             },
         )
 
-        self.cache[Member].add(Member(**request))
+        # self.cache[Member].add(Member(**request))
+        # This should fire a gw event?
```
Comment (Toricane) [inline interactions/api/http/guild.py]:
```suggestion
```

**Review by Toricane — CHANGES_REQUESTED** [review]
Few suggestions

**Review by Toricane — APPROVED** [review]
LGTM!

**Comment by Damego** [comment]
> LGTM!

LBTM xD

**Pull request merged**